job/cache.go

```diff
@@ -115,6 +115,18 @@ func (c *MemoryJobCache) Delete(id string) {
 	c.jobs.Lock.Lock()
 	defer c.jobs.Lock.Unlock()
 
+	j := c.jobs.Jobs[id]
+	if j == nil {
+		// TODO
+		//return nil, JobDoesntExistErr
+		return
+	}
+
+	// TODO return error
+	c.jobs.Lock.Unlock()
```

Comment:
@ajvb Note that if just the delete is in a goroutine then of course the locking done in (*MemoryJobCache).Delete() won't protect it; the `defer unlock` will run as this method is returning, regardless of what's going on in whatever goroutines it launched.